### PR TITLE
Fixes #37300 - Change I18n loading from import to require

### PR DESCRIPTION
**Pull request by MariaAga**
To avoid this error when updating babel-presets in foreman-js
```
Uncaught (in promise) ChunkLoadError: Loading chunk react-intl/locale/en failed.
(error: http://localhost:3808/webpack/react-intl/locale/en.js)
    at __webpack_require__.f.j (jsonp chunk loading:27:1)
    at ensure chunk:6:1
    at Array.reduce (<anonymous>)
    at __webpack_require__.e (ensure chunk:5:1)
    at webpackAsyncContext ([request] namespace object:69:1)
    at IntlLoader._callee$ (I18n.js:17:30)
    at tryCatch (I18n.js:2:1)
    at Generator.<anonymous> (I18n.js:2:1)
    at Generator.next (I18n.js:2:1)
    at asyncGeneratorStep (I18n.js:2:1)
```

Trying to update babel presets again. 
Tested with theforeman 12.2.2 as that one has the updates that we had to revert.
Updating babel/presets caused dynamic imports to be in their own chunk, which caused them to not be loaded, using dynamic require does not create the access chunks.

### CODE REVIEW

File: webpack/assets/javascripts/react_app/common/I18n.js [hunk 1]
```diff
@@ -61,7 +59,8 @@ const mergeLocaleData = locale => {
     Object.entries(translations).forEach(([source, translated]) => {
       if (
         result[source] === undefined ||
-        (result[source] === [''] && translated !== [''])
+        (JSON.stringify(result[source]) === JSON.stringify(['']) &&
+          JSON.stringify(translated) !== JSON.stringify(['']))
```
Comment (adamruzicka) [inline webpack/assets/javascripts/react_app/common/I18n.js]: Why is this necessary?

Comment (ofedoren) [reply webpack/assets/javascripts/react_app/common/I18n.js]: Maybe because in JS `[''] === [''] // false` ?..

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/common/I18n.js]: /me sighs

Of course. This is evaluated each time the application is loaded, correct? Which may or may not mean every time the user navigates to a page?

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: What Oleh said... My code editor was angry about this so I fixed it here. I put a console.log here to check and it ran 4k times in opening the login page.

Comment (ekohl) [reply webpack/assets/javascripts/react_app/common/I18n.js]: That makes me think of https://www.destroyallsoftware.com/talks/wat

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: (it took me a few years to agree to move from C to JS after seeing this video for the first time)

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/common/I18n.js]: > it ran 4k times in opening the login page.

That's quite a lot. I'm pretty sure there's a more efficient way of determining whether an array contains a single element (which is an empty string) rather than dumping both sides into json and comparing those. If this got run only occasionally then I'd be willing to let this slide, but this is too frequent

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: I can check the first element `result[source].?[0] === ''`?

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: BTW its runs for every string that should be translated

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/common/I18n.js]: > I can check the first element result[source].?[0] === ''?

Or length and first element?

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: `result[source].?[0].length === 0`?

Comment (adamruzicka) [reply webpack/assets/javascripts/react_app/common/I18n.js]: I originally meant something like `result[source]?.length === 1 && result[source][0] === ''`, but whatever works

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js]: most of them look like they have result[source]?.length === 1, updated the code

File: webpack/assets/javascripts/react_app/common/I18n.js [hunk 2]
```diff
@@ -14,19 +16,15 @@ class IntlLoader {
 
   async init() {
     await this.fetchIntl();
-    const localeData = await import(
```
Comment (ofedoren) [inline webpack/assets/javascripts/react_app/common/I18n.js:17]: Since I have almost 0 knowledge about JS world, I searched "difference between import and require in JS" and looking at the answers it seems that this file shouldn't be working at first place?..

I mean, they say `import` can't be used conditionally and anywhere except the beginning of a file. Also, they say that `require` is synchronous, so we don't need `async/await` here?..

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/common/I18n.js:17]: You're right, removed the await

**Review by ofedoren — APPROVED** [review]
